galsim/utilities.py

```diff
@@ -418,6 +418,103 @@ def _gammafn(x):
+        >>> gal = galsim.Gaussian(sigma=2.8)
+        >>> gal_pix = galsim.Convolve([gal,galsim.Pixel(scale=1.0)])
+        >>> img = gal_pix.drawImage(gal_pix,scale=1.0*n,method='no_pixel')
+        >>> im_list, offsets = galsim.utilities.deInterleaveImage(gal_pix,N=n)
+        >>> for im in im_list:
+        >>>     im.applyNonlinearity(lambda x: x-0.01*x**2) #detector effects
+        >>> img_new = galsim.utilities.interleaveImages(im_list,N=n,offsets)
+
+    @param image             Input image from which lower resolution images are extracted.
+    @param N                 Number of images extracted in either directions. It can be of type
+                             'int' if equal number of images are extracted in both directions or a
+                             list or tuple of two integers, containing the number of images in x
+                             and y directions respectively.
+    @param conserve_flux     Should the routine output images that have, on average, same total
+                             pixel values as the input image (True) or should the pixel values
+                             summed over all the images equal the sum of pixel values of the input
+                             image (False)? [default: False]
+    @param suppress_warnings Suppresses the warnings about the pixel scale of the output, if True.
+                             [default: False]
+
+    @returns a list of images and offsets to reconstruct the input image using 'interleaveImages'.
+    """
+
+    if isinstance(N,int):
+        n1,n2 = N,N
+    elif hasattr(N,'__iter__'):
+        if len(N)==2:
+            n1,n2 = N
+        else:
+            raise TypeError("'N' has to be a list or a tuple of two integers")
+        if not (isinstance(n1,int) and  isinstance(n2,int)):
+            raise TypeError("'N' has to be of type int or a list or a tuple of two integers")
+    else:
+        raise TypeError("'N' has to be of type int or a list or a tuple of two integers")
+
+    if not isinstance(image,galsim.Image):
+        raise TypeError("'image' has to be an instance of galsim.Image")
+
+    y_size,x_size = image.array.shape
+    if x_size%n1 or y_size%n2:
+        raise ValueError("The value of 'N' is incompatible with the dimensions of the image to "+
+                         +"be 'deinterleaved'")
+
+    im_list, offsets = [], []
+    for i in xrange(n1):
+        for j in xrange(n2):
+            # The tricky part - going from array indices to Image coordinates (x,y)
+            # DX[i'] = -(i+0.5)/n+0.5 = -i/n + 0.5*(n-1)/n
+            #    i  = -n DX[i'] + 0.5*(n-1)
+            dx,dy = -(i+0.5)/n1+0.5,-(j+0.5)/n2+0.5
+            offset = galsim.PositionD(dx,dy)
+            img_arr = image.array[j::n2,i::n1].copy()
+            img = galsim.Image(img_arr)
+            if conserve_flux is True:
+                img *= n1*n2
+            im_list.append(img)
+            offsets.append(offset)
+
+    wcs = image.wcs
```

Comment:
I see these WCS checks, but I don't see bounds checks.  I don't think that if the original image has some non-trivial bounds, that the new one will respect that.